src/httpclient.c

```diff
@@ -114,7 +118,26 @@ void httpclient_timeout_config(
     if (curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT_MS, cto) != CURLE_OK) {
         corto_error("Failed to set CURLOPT_CONNECTTIMEOUT_MS.");
     }
+}
+
+void httpclient_auth_config(
+    CURL *curl)
+{
+    corto_string user = httpclient_get_user();
+    corto_string password = httpclient_get_password();
 
+    if (!user || !password) {
+        /* Auth Not configured */
+        return;
+    }
+
+    if (curl_easy_setopt(curl, CURLOPT_USERNAME, user) != CURLE_OK) {
+        corto_error("Failed to set CURLOPT_USERNAME.");
```

Comment:
Would be preferable to use `corto_throw`, and return a non-zero return code so that the callee of this function can see if it failed or not.

Reply:
Good Idea.